File: src/SamuraiServer/SamuraiServer.Data/Models/GameState.cs
using System;
using System.Collections.Generic;

namespace SamuraiServer.Data
{
    public class GameState
    {
        public Guid Id { get; set; }

        public string Name { get; set; }  // NOTE: what does this represent? <csainty: Lets the user name a game so they can tell a friend which to join. Could use invites instead>

        public List<GamePlayer> Players { get; set; }

        public int Turn { get; set; }

        public bool Started { get; set; }

        public Guid MapId { get; set; }

        public List<Guid> PlayerOrder { get; set; } 

        public GameState()
        {
            Players = new List<GamePlayer>();
            PlayerOrder = new List<Guid>();
        }
    }
}

File: src/Samurai.Client.JS/models/gameState.js
function GameState(opts) {
  this.initialize(opts);
}

GameState.prototype.initialize = function (opt) {
  var key = null,
      default_args = {
        //public Guid Id
        Id : '',
        Name : '',  // NOTE: what does this represent? <csainty: Lets the user name a game so they can tell a friend which to join. Could use invites instead>
        //List<GamePlayer>
        Players : [],
        Turn : 0,
        Started : false,
        //public Guid
        MapId : '',
        //List<Guid>
        PlayerOrder : [],
      };

  opt || (opt = default_args);

  for(key in default_args) {
    if(typeof opt[key] == "undefined") opt[key] = default_args[key];
  }

  // opt[] has all the data - user provided and optional.
  for(key in opt) {
    this[key] = opt[key];
  }
};